src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -0,0 +1,462 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.NamingStyles;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+using static Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SymbolSpecification;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionSetSources
+{
+    public class DeclarationNameCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public DeclarationNameCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider()
+        {
+            return new DeclarationNameCompletionProvider();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void NameWithOnlyType1()
+        {
+            var markup = @"
+public class C
+{
+    C $$
+}
+";
+            await VerifyItemExistsAsync(markup, "C", glyph: (int)Glyph.PropertyPublic);
+            await VerifyItemExistsAsync(markup, "c", glyph: (int)Glyph.FieldPublic);
+            await VerifyItemExistsAsync(markup, "GetC", glyph: (int)Glyph.MethodPublic);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void AsyncTaskOfT()
+        {
+            var markup = @"
+public class C
+{
+    async Task<C> $$
+}
+";
+            var workspace = WorkspaceFixture.GetWorkspace();
+            var oldOptions = workspace.Options;
+            try
+            {
+                var newOptions = oldOptions.WithChangedOption(new OptionKey(SimplificationOptions.NamingPreferences, LanguageNames.CSharp), AsyncMethodsEndWithAsync());
```

Comment:
Ah, I actually did hardcode it in but forgot to update this test--will remove the naming rules added by the test.